scripts/tasks_retrieval/uniprot_keyword_tasks_creation.py

```diff
@@ -0,0 +1,171 @@
+    category_dict = defaultdict(lambda: defaultdict(set))
+
+    for gene_protein in gene_proteins:
+        gene_symbol = gene_protein["genes"][0]["geneName"]["value"]
+
+        for kw in gene_protein.get("keywords", []):
+            category = kw["category"]
+            name = kw["name"]
+            category_dict[category][gene_symbol].add(name)
+    return category_dict
+
+
+def make_gene_kw_df(gene_kw_map: dict[str, set]) -> pd.DataFrame:
+    all_keywords = sorted({kw for keywords in gene_kw_map.values() for kw in keywords})
+
+    rows = []
+    for gene, keywords in gene_kw_map.items():
+        row = [1 if kw in keywords else 0 for kw in all_keywords]
+        rows.append([gene] + row)
+
+    return pd.DataFrame(rows, columns=["Gene"] + all_keywords).set_index("Gene")
+
+
+def download_and_load_json_gz(url: str) -> dict:
+    """
+    Download and gunzip a json from a url.
+
+    Args:
+    ----
+        url (str): url to download
+
+    Returns:
+    -------
+        dict: contents of json.gz as a dict
+
+    """
+    response = requests.get(url)
+    response.raise_for_status()
+    compressed_data = BytesIO(response.content)
+    with gzip.GzipFile(fileobj=compressed_data) as gz:
+        json_data = json.load(gz)
+
+    return json_data
+
+
+def get_uniprot_human_protein_features(
+    file: str | None = None, allow_downloads: bool = False
+) -> dict:
+    """Get UniProt Human Proteins data from file or from server."""
+    if file and Path(file).exists():
+        with open(file) as f:
+            return json.load(f)
+    url = "https://rest.uniprot.org/uniprotkb/stream?compressed=true&download=true&format=json&query=%28*%29+AND+%28model_organism%3A9606%29+AND+%28reviewed%3Atrue%29"
+    if allow_downloads:
+        uniprot_features = download_and_load_json_gz(url)
+    if file:
+        with open(file, "w") as f:
+            json.dump(uniprot_features, f)
+
+    return uniprot_features
+
+
+@click.command()
+@click.option(
+    "--allow-downloads",
+    "-l",
+    type=click.BOOL,
+    help=f"download files directly from {GENE_SYMBOL_URL}, use this option only if you trust the URL.",
+    default=False,
+)
+@click.option(
+    "--input-file",
+    type=click.STRING,
+    help="The path to the yaml file with the gene names. If omitted, `allow-downloads` must be True",
+    default=None,
+)
+@click.option(
+    "--main-task-directory",
+    "-m",
+    type=click.STRING,
+    help="The task root directory.  Will not be created.",
+    default="./tasks",
+)
+@click.option(
+    "--task-name",
+    "-n",
+    type=click.STRING,
+    multiple=True,
+    help="Name for the task based on UniProt keyword category. Must be from this list:"
+    "  ['Biological process', 'Cellular component', 'Coding sequence diversity', 'Disease', "
+    "    'Domain', 'Ligand', 'Molecular function', 'PTM', 'Technical term']"
+    "Can be multiply defined. Defaults to creating all of the possible keyword tasks.",
+    default=[
```

Comment:
The default should be the subset that we want. So without "Biological process," for example